src/allocator/memserver_allocator.cpp

```diff
@@ -233,6 +299,8 @@ void Memserver_Allocator::destroy_region(uint64_t regionId) {
                         message.str().c_str());
     }
 
+    remove_heap_from_list(regionId);  
```

Comment:
This needs to be synchronized with delayed free thread..

Reply:
I think like in nvmm, if we are adding a conditional variable while destroy_region is in progress will do.  Because in these two functions, what we are accessing (and which is global) is only teh thread_array which will be modified only by constructor and destructor. so there we dont be ahving any issues.